server/channels/app/channel_bookmark_test.go

```diff
@@ -0,0 +1,422 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func find_bookmark(slice []*model.ChannelBookmarkWithFileInfo, id string) *model.ChannelBookmarkWithFileInfo {
+	for _, element := range slice {
+		if element.Id == id {
+			return element
+		}
+	}
+	return nil
+}
+
+func createBookmark(name string, bookmarkType model.ChannelBookmarkType, channelId string, fileId string) *model.ChannelBookmark {
+	bookmark := &model.ChannelBookmark{
+		ChannelId:   channelId,
+		DisplayName: name,
+		LinkUrl:     "https://mattermost.com",
+		Type:        bookmarkType,
+		Emoji:       ":smile:",
+		FileId:      fileId,
+	}
+
+	return bookmark
+}
+
+func TestCreateBookmark(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	t.Run("create a channel bookmark", func(t *testing.T) {
+		th.Context.Session().UserId = th.BasicUser.Id // set the user for the session
+
+		bookmark1 := createBookmark("Link bookmark test", model.ChannelBookmarkLink, th.BasicChannel.Id, "")
+		bookmarkResp, err := th.App.CreateChannelBookmark(th.Context, bookmark1, "")
+		assert.Nil(t, err)
```

Comment:
We shouldn't use `assert` here as `bookmarkResp` can be `nil` if there is an error. It would cause panics. There are similar occurrences throughout the file, can you double check?